src/AbstractDoctrineAnnotationFixer.php

```diff
@@ -29,6 +29,9 @@
  */
 abstract class AbstractDoctrineAnnotationFixer extends AbstractFixer implements ConfigurableFixerInterface
 {
+    /**
+     * @var array<int, array{classIndex: int, token: Token, type: string}>
```

Comment:
can we say what is the array index? or it is not relevant?

I'm wondering if it's `[foo, bar]` or `[4 => foo, 7 => bar]`. if first, I way rather use `string[]` over `array<int, string>`

Reply:
Could you see https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6293#discussion_r820157214 and let me know whether that makes sense to you?

If it was `[foo, bar]`, I'd use `list<string>`, as `string[] === array<array-key, string>`.

Reply:
> Static analysis tools interpret CaseAnalysis[] as array<array-key, CaseAnalysis>, which tells you nothing about keys.

can you share some references?
I cannot find anything about `list<...>` in PSRs (I know you mentioned "some tools", but I would prefer to have links)

PHP itself has pseudotype `iterable` and concept of `list` (yet not a type... - vide array_is_list method). maybe we should have a guard in `tests/AutoReview/` to prevent usage of `...[]` globally for the project? then whoever would be introducing `...[]` notation by habit, we would prevent that automatically.

Reply:
> can you share some references?

Psalm docs on [lists](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#lists) and [`[]` syntax](https://psalm.dev/docs/annotating_code/type_syntax/array_types/#phpdoc-syntax), here is an example in [Psalm](https://psalm.dev/r/406314c3d2) and [PHPStan](https://phpstan.org/r/3f9a02f5-8747-4cda-a8bd-4095928fc737).

> maybe we should have a guard in tests/AutoReview/ to prevent usage of ...[] globally for the project?

I would be all for that.

Reply:
> maybe we should have a guard in `tests/AutoReview/` to prevent usage of `...[]` globally for the project?

Could even be a new fixer ;)

Reply:
Not sure whether you ask me or @keradus? As for me, this PR is ready :)